lib/charms/opensearch/v0/helper_conf_setter.py

```diff
@@ -289,10 +288,9 @@ def replace(
         if output_type in [OutputType.console, OutputType.all]:
             logger.info(data)
 
-        if output_file is None:
-            output_file = config_file
-
-        with open(output_file, "w") as f:
+        if output_file and output_file != config_file:
```

Comment:
```suggestion
        if output_file != config_file:
```

Reply:
@Mehdi-Bendriss path has been already defined. So the question is: are we writing to a different file, or the same?

Check L274